src/components/Chart/TimeBarsAnnotations.js

```diff
@@ -107,21 +121,30 @@ export const XAnnotation = ({
     labelText ? labelGauger(labelText) : 0,
     valueText ? valueGauger(valueText) : 0
   )
-  const x1 = isRange ? xCalc(annotation.x1) : xCalc(annotation.x)
+  const x1 = annotation.leftLabel
+    ? 0
+    : isRange
+    ? xCalc(annotation.x1)
+    : xCalc(annotation.x)
   const x2 = isRange
     ? xCalc(annotation.x2) + barWidth
     : x1 + Math.max(barWidth, 8)
 
   let textAnchor = 'middle'
-  let tx = x1 + (x2 - x1) / 2
-  if (x1 + (x2 - x1) / 2 + textSize / 2 > width) {
-    textAnchor = 'end'
-    tx = x2
-    if ((isRange ? x2 : x1) - textSize < 0) {
-      textAnchor = 'start'
-      tx = x1
+  let tx = annotation.leftLabel ? 0 : x1 + (x2 - x1) / 2
+  if (annotation.textAlignment === undefined) {
+    if (x1 + (x2 - x1) / 2 + textSize / 2 > width) {
```

Comment:
i'd put add an else clause to this `if`
```
else {
  textAnchor = 'middle'
}
```
here.
Then you cover all the cases where textAlignment is undefined here and can ignore the undefined case altogether in l. 145